src/d3a/d3a_core/export.py

```diff
@@ -386,6 +389,56 @@ def _plot_ess_soc_history(self, storage_list: list, subdir: str, root_name: str)
         output_file = os.path.join(plot_dir, 'ess_soc_history_{}.html'.format(root_name))
         BarGraph.plot_bar_graph(barmode, title, xtitle, ytitle, data, output_file)
 
+    def supply_demand_curve(self, area, subdir):
+        """
+        Wrapper for _supply_demand_curve
+        """
+        new_subdir = os.path.join(subdir, area.slug)
+        self._supply_demand_curve(new_subdir, area)
+        for child in area.children:
+            if child.children:
+                self.supply_demand_curve(child, new_subdir)
+
+    def _supply_demand_curve(self, subdir: str, area: Area):
+
+        for i in area.past_markets:
```

Comment:
I am not a big fan of using single-character variable names, because it decreases the readability.
How about renaming all single-character variables in this function, e.g. "i" --> "past_market" ?